### PR TITLE
`install-argo-cd.sh dev-$FNHSNAME MINE`

### PR DESCRIPTION
**Pull request by alsuren**
Use `install-argo-cd.sh dev-$FNHSNAME --mine` to switch argocd to the branch you're currently on (assuming that you have already made a PR).

### CODE REVIEW

File: infrastructure/scripts/install-argo-cd.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -eu
 cd $(dirname "$0")
 
 ENVIRONMENT="${1:?"Please specify you environment name as the first parameter, e.g. dev-jane"}"
+BRANCH="${2:?"Please specify the branch you want to deploy, e.g. master or pr-mybranch or --mine"}"
```
Comment (frigus02) [inline infrastructure/scripts/install-argo-cd.sh:8]: Could we default this to master? Or default to whatever the overlay variables file has?

File: infrastructure/scripts/install-argo-cd.sh [hunk 2]
```diff
@@ -14,6 +16,13 @@ if [ "$ENVIRONMENT" != "$CURRENT_CONTEXT" ]; then
 	exit 1
 fi
 
+if [ "x$BRANCH" = "x--mine" ]; then
```
Comment (frigus02) [inline infrastructure/scripts/install-argo-cd.sh:19]: What's the "x" protecting against? It rings a bell, but I forgot.

Comment (alsuren) [reply infrastructure/scripts/install-argo-cd.sh:19]: it's a portable shell thing from the days of HP unix or somesuch https://unix.stackexchange.com/a/136632 . I think I picked it up off my boss in my work experience after my first year of uni.  In my head I still think of `[` as an alias for the `test` command, and am scared of `[ "$BRANCH" = "--mine" ]` expanding strangely and doing the wrong thing. I don't think any shells do this anymore though.

File: infrastructure/README.md [hunk 3]
```diff
@@ -213,6 +213,12 @@ as opposed to sharing a staging environment.
 
    If you want to see the frontend app browse to <https://fnhs-dev-$FNHSNAME.westeurope.cloudapp.azure.com>.
 
+   If you want to switch branches in argocd, re-run install-argo-cd.sh specifying the branch name, or --mine.
+
+   ```bash
+   ./infrastructure/scripts/install-argo-cd.sh dev-$FNHSNAME --mine
```
Comment (frigus02) [inline infrastructure/README.md:219]: Should this say:
```suggestion
   ./infrastructure/scripts/install-argo-cd.sh dev-$FNHSNAME MINE
```

File: .github/workflows/frontend-branch.yaml [hunk 4]
```diff
@@ -80,10 +81,10 @@ jobs:
           n=0
           until [ $n -ge 5 ]
           do
-            # TODO: also push to pr-$GITHUB_ACTOR branch. Can do this as a force-push at the end?
             git push origin "HEAD:pr-$GITHUB_HEAD_REF" && break
             n+=1
-            git pull --rebase origin pr-$GITHUB_HEAD_REF
+            git fetch origin pr-$GITHUB_HEAD_REF
+            git merge origin/master --strategy=recursive --strategy-option=ours
```
Comment (frigus02) [inline .github/workflows/frontend-branch.yaml:87]: Do we need to merge the branch `pr-$GITHUB_HEAD_REF` as well? E.g.:

```suggestion
            git merge origin/master origin/pr-$GITHUB_HEAD_REF --strategy=recursive --strategy-option=ours
```

Comment (alsuren) [reply .github/workflows/frontend-branch.yaml:87]: ugh. well spotted.

File: .github/workflows/frontend-branch.yaml [hunk 5]
```diff
@@ -80,10 +81,10 @@ jobs:
           n=0
           until [ $n -ge 5 ]
           do
-            # TODO: also push to pr-$GITHUB_ACTOR branch. Can do this as a force-push at the end?
             git push origin "HEAD:pr-$GITHUB_HEAD_REF" && break
             n+=1
-            git pull --rebase origin pr-$GITHUB_HEAD_REF
+            git fetch origin pr-$GITHUB_HEAD_REF
+            git merge origin/pr-$GITHUB_HEAD_REF --strategy=recursive --strategy-option=ours
```
Comment (frigus02) [inline .github/workflows/frontend-branch.yaml:87]: (I had a longer lunch, which is why I'm still working.)

Can this still lead to unwanted results in a very specific scenario?

| Time | pr-1 (kratos build) | pr-2 (kratos build) | pr-2 (frontend build) |
| ---: | --- | --- | --- |
| 1 | | Checkout the correct Deployments repo branch | |
| 2 | Merged. Updates deployments repo master branch. | | |
| 3 | | | Checkout the correct Deployments repo branch |
| 4 | | kustomize edit; git commit | |
| 5 | | git push | |
| 6 | | | kustomize edit; git commit |
| 7 | | | git push (ERROR) |
| 8 | | | git merge origin/pr-2 -X ours (prefers master version of kratos folder?) |

We could prevent those by re-doing all steps from "Checkout the correct Deployments repo branch" again.

Comment (frigus02) [reply .github/workflows/frontend-branch.yaml:87]: Even if that's the case, we might still want to merge it like this. It's way better than before.

Comment (alsuren) [reply .github/workflows/frontend-branch.yaml:87]: I guess we will have to see. At least we will have the full git history (without rebases) to help us debug it :D

**Pull request merged**